api/workOrder/workOrderRouter.js

```diff
@@ -0,0 +1,494 @@
+          );
+        } else {
+          res.status(400).json({ message: 'workOrder already exists' });
+        }
+      });
+    } catch (e) {
+      console.error(e);
+      res.status(500).json({ message: e.message });
+    }
+  } else {
+    res.status(404).json({ message: 'WorkOrder missing' });
+  }
+});
+/**
+ * @swagger
+ * /workOrder:
+ *  put:
+ *    summary: Update a workOrder
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - workOrder
+ *    requestBody:
+ *      description: WorkOrder object to to be updated
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/WorkOrder'
+ *    responses:
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        $ref: '#/components/responses/NotFound'
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: workOrder created
+ *                workOrder:
+ *                  $ref: '#/components/schemas/WorkOrder'
+ */
+router.put('/', authRequired, function (req, res) {
+  const workOrder = req.body;
+  if (workOrder) {
+    const id = workOrder.id || 0;
+    WorkOrders.findById(id)
+      .then(
+        WorkOrders.update(id, workOrder)
+          .then((updated) => {
+            res
+              .status(200)
+              .json({ message: 'workOrder created', workOrder: updated[0] });
+          })
+          .catch((err) => {
+            res.status(500).json({
+              message: `Could not update workOrder '${id}'`,
+              error: err.message,
+            });
+          })
+      )
+      .catch((err) => {
+        res.status(404).json({
+          message: `Could not find workOrder '${id}'`,
+          error: err.message,
+        });
+      });
+  }
+});
+/**
+ * @swagger
+ * /workOrder/{id}:
+ *  delete:
+ *    summary: Remove a workOrder
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - workOrder
+ *    parameters:
+ *      - $ref: '#/components/parameters/workOrderId'
+ *    responses:
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        $ref: '#/components/responses/NotFound'
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: WorkOrder 3 was deleted.
+ *                workOrder:
+ *                  $ref: '#/components/schemas/WorkOrder'
+ */
+router.delete('/:id', authRequired, function (req, res) {
```

Comment:
Attempting to do a delete via Insomnia gave me a really nasty error message, not sure if this is just for me or if it's happening to other people.

Reply:
You're right. I was able to delete in my tests, but, of course, that's not a real database environment. You are right that we need some sort of cascade or allowing for null on a delete. I will work on that.

Reply:
Fixed with the `bug-fix/database-foreign-key-fix` branch.